### ISSUE TITLE
wrapper fails due to missing functions in gir file

### ISSUE THREAD

**Issue by hyOzd** [gtkd-developers/GtkD#182]
Hi I'm trying to run the wrapper under Linux Mint 18. I'm having this error:

```
copying file [./paths.d] to [./out/src/gtkc/paths.d]
copying file [./Loader.d] to [./out/src/gtkc/Loader.d]
utils.WrapError.WrapError@./APILookupGio.txt(665): unknown function keyfile_settings_backend_new
```

This function exists in `gsettingsbackend.h` file but it's not mentioned in `/usr/share/gir-1.0/Gio-2.0.gir`. I think this is the problem. Does this mean this 'gir' file isn't up to date? What should I do? Should I try to fix the gir file (re-generate it myself??) or just remove the problematic definitions from APILookup file until it succeeds?

Note: I can use already generated wrappers in this repository but I'm planning to have a shot at creating wrappers for another gnome library. Being inexperienced in both D and gtk libraries I thought I should at least be able to re-generate existing wrappers.

**Comment by MikeWey** [comment]
The wrapper currently doesn't handle older versions of the gir files that well. It should work well with the versions mentioned at the bottom of the README. If Linux Mint hat those versions installed you could probably just remove the problematic definitions for now.

**Comment by hyOzd** [comment]
It seems libraries are up to date but the gir files which is distributed in a separate package called `libgirepository1.0-dev` are not. 

I will try removing stuff then. Thanks for reply.

**Comment by hyOzd** [comment]
Ok, I got it working finally. I disabled a bunch of stuff from the apilookup files : ) Here is what I've learned so far:

- Run ./gtkwrap, see error, disable the relevant line from lookup file run ./gtkwrap again
- src/ folder had to be removed for a clean operation
- Except "src/cairo", one should keep it because it is skipped by wrapper (hardcoded in wrapper source)
- Most errors (missing functions etc) are evident during the wrapper run, but wrapper will successfully generate some files that doesn't compile due to missing symbols. These should be also disabled from the api lookup files.
